core/src/main/java/org/geysermc/geyser/session/GeyserSession.java

```diff
@@ -2014,23 +2014,30 @@ public float getEyeHeight() {
      * @param strength value between 0 and 1
      */
     public void updateRain(float strength) {
-        this.raining = strength > 0;
+        boolean transition = isRaining() ^ strength > 0; // Rain is starting/stopping
```

Comment:
reason for `^` instead of `!=` ?

Reply:
and add a comment about why we aren't doing `this.rainStrength != strength` just to avoid mistakes in the future

Reply:
Updated code